2025-03 package updates

kmonahan opened the pull request:
Minor and package updates, plus updates to @swc/cli, @storybook/addon-webpack5-compiler-swc, webpack-cli, react-refresh, react, and react-dom.

Most noteworthy update is to React and React DOM. The previously deprecated global JSX namespace has now been removed, so you definitely have to import it if you're using React with TypeScript, and the syntax for rendering a React component on a page has changed slightly. Even still, changes are pretty minor here.

kmonahan committed:
Minor and patch version updates

kmonahan committed:
Update @swc/cli to 0.6.0 and @storybook/addon-webpack5-compiler-swc to 3.0.0

kmonahan committed:
Update webpack-cli to 6.0.1

kmonahan committed:
Update react-refresh to 0.16.0

kmonahan committed:
Update react and react-dom to v19

tomealter commented:
@kmonahan Gave this a look over, the theme builds fine initially. Since you called out the React changes, I tested the build there too and I did notice an issue.
If you make any changes to the `source/07-react/Example/Example.tsx`, the watch build throws the following error:
`ERROR in TS1: TSC compiler crashed: Debug Failure. False expression. Error: Debug Failure. False expression.`

coreylafferty reviewed (changes requested):
Tommy requests changes.

kmonahan committed:
Fix `TSC compiler crashed: Debug Failure. False expression.` error

Error occurred any time a TSX file was edited. Related to https://github.com/microsoft/TypeScript/issues/60692. TypeScript doesn't like exports of globals, even in JS files it shouldn't be checking anyway.

kmonahan commented:
@tomealter Looks like the issue was related to https://github.com/microsoft/TypeScript/issues/60692 and the same workaround worked for me locally. Please retest and see if error is fixed on your end as well!

coreylafferty committed:
Merge branch '5.x-RC' into 2025-03-package-updates

coreylafferty committed:
updates lock file

coreylafferty commented:
@kmonahan I went to test this and install failed saying the lock file was not in sync with the package file. Not sure if that was overlooked or if something new has been recently released -- but I created this PR for the lock changes. Can you confirm these look ok? If so, I'll more forward with merging both PRs.  

https://github.com/forumone/gesso/pull/936

coreylafferty committed:
Merge pull request #936 from forumone/2025-03-package-updates-lock-file-change

updates lock file
